### PR TITLE
Preparing for Jest 20.x

### PR DESCRIPTION
**Pull request by orta**
* Updates the Jest dependency
* Fixes a few annoyances
* The Jest update should fix a lot of the JS + Babel issues
* Hopefully I can persuade a patch release with https://github.com/facebook/jest/pull/3592 or it'll not start up with older versions of Jest

Maybe until then I can read the `package.json` for Jest for the exact version in the meantime, BRB

### CODE REVIEW

File: src/JestExt.ts [hunk 1]
```diff
@@ -121,9 +121,10 @@ export class JestExt {
     }
 
     private getSettings() {
+
         this.jestSettings.getConfig(() => {
-            if (this.jestSettings.jestVersionMajor < 18) {
-                vscode.window.showErrorMessage('This extension relies on Jest 18+ features, it will work, but the highlighting may not work correctly.');
+            if (this.jestSettings.jestVersionMajor < 20) {
+                vscode.window.showErrorMessage('This extension relies on Jest 20+ features, it continue to work, some features may not work correctly.');
```
Comment (bookman25) [inline src/JestExt.ts]: `it continues to work`?

File: tests/JestExt.test.ts [hunk 2]
```diff
@@ -24,13 +24,13 @@ describe('JestExt', () => {
             jestVersionMajor: 17,
         }));
         new JestExt(projectWorkspace, channelStub, {});
-        expect(window.showErrorMessage).toBeCalledWith('This extension relies on Jest 18+ features, it will work, but the highlighting may not work correctly.');
+        expect(window.showErrorMessage).toBeCalledWith('This extension relies on Jest 20+ features, it continue to work, some features may not work correctly.');
```
Comment (bookman25) [inline tests/JestExt.test.ts]: Wonder if we should just switch this to a snapshot.  That way we don't have the duplicated text and can just verify the snapshot.

Comment (orta) [inline tests/JestExt.test.ts]: 👍 doing this - thanks

File: tests/JestExt.test.ts [hunk 3]
```diff
@@ -24,13 +25,13 @@ describe('JestExt', () => {
             jestVersionMajor: 17,
         }));
         new JestExt(projectWorkspace, channelStub, {});
-        expect(window.showErrorMessage).toBeCalledWith('This extension relies on Jest 18+ features, it will work, but the highlighting may not work correctly.');
+        expect(window.showErrorMessage.mock.calls).toMatchSnapshot();
     });
 
-    it('should not show error message if jest version is 18', () => {
+    it.skip('should not show error message if jest version is 20', () => {
```
Comment (orta) [inline tests/JestExt.test.ts]: I know, I know - but not got all day on this 💯

**Comment by orta** [comment]
screen shot 2017-05-17 at 16 38 53

This is an old CRA app with Jest 17 👍

**Pull request merged**